### DISCUSSION TITLE
Weird slug behavior. Uri's are not always unique

### DISCUSSION THREAD

**Discussion by jymden** [statamic/cms#7448] [Troubleshooting]
I'm not quite certain how this is supposed to work. But in the control panel I can create an entry in a collection with the same name as another entry in the same collection. Statamic will then generate exactly the same slug and uri with no complaints. But if I edit the entry, and try to save it, Statamic will tell me that the uri is in use. 

So there are two different behaviors in the CP.  So I'm wondering:

1. Is this expected behavior?
2. Is it a bug?
3. Is it something in my configuration? (a clean install of Statamic, with Eloquent Driver acts the same) 
4. How can I guarantee unique uri's?

**Comment by jasonvarga** [comment]
What `route` are you using in the collection?

**Comment by jymden** [comment]
Route is either` {{ parent_uri }}/{{ slug }}` or just `collection_name/{{slug}} `
I'm just a bit confused about the behavior that I could create an entry with duplicate slug/uri but I can't edit and save one.

**Reply by jasonvarga** [reply]
I can't replicate that behavior, which is why I was asking about the route. 

Can you provide a github repo for us to look at?

**Reply by jymden** [reply]
Thank you Jason. Just realized that I made a mistake describing the problem to you, I'm sorry. The collections with `collection_name/{{slug}}` routes actually does seem to work. The ones that doesn't work are configured like this:
`route: '{{ depth == 1 ?= "/collection_name" }}/{{ parent_uri }}/{{ slug }}'` because they need to be prefixed with the collection name and still have the hierarchy retained in the uri. So with this route config, the root page get "/collection_name" as uri instead of '/' and any other pages get collection_name/slug. And any children get /collection_name/parent_slug/slug etc,.

So it seems that with a route configured like that the uri check is just skipped when saving making it possible to have duplicate uri's in the same collection, but only the first time. If you edit the page and try to save it again it doesn't validate.
